Data/SQLite/src/SessionImpl.cpp

```diff
@@ -205,16 +205,20 @@ void SessionImpl::open(const std::string& connect)
 
 void SessionImpl::close()
 {
-	if (_pDB) {
+	if (_pDB) 
+	{
 		int result = 0;
 		int times = 50;
-		do {
+		do 
+		{
 			result = sqlite3_close(_pDB);
-			if (result == SQLITE_BUSY) {
+			if (result == SQLITE_BUSY) 
```

Comment:
Looping limited number of times prevents endless loop but does not necessarily fix the root cause for the busy state. If this is happening, something is [still opened](http://www.sqlite.org/c3ref/close.html) - likely a prepared statement, which should be finalized (or should not be lingering to start with).